src/ci_workflow/ci_check_package.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import json
+import os
+
+from ci_workflow.ci_check import CiCheckSource
+from system.properties_file import PropertiesFile
+
+
+class CiCheckPackage(CiCheckSource):
+    def __init__(self, component, git_repo, target, args=None):
+        super().__init__(component, git_repo, target, args)
+        self.properties = self.__get_properties()
+
+    @property
+    def package_json_path(self):
+        return os.path.join(self.git_repo.working_directory, "package.json")
+
+    def __get_properties(self):
+        with open(self.package_json_path, "r") as f:
+            return PropertiesFile(CiCheckPackage.__flattenDict(json.load(f)))
+
+    # https://gist.github.com/higarmi/6708779
+    @classmethod
+    def __flattenDict(cls, d, result=None, index=None, parent_key=None):
+        if result is None:
+            result = {}
+        if isinstance(d, (list, tuple)):
```

Comment:
Can we change `d` to actually `dict` or `dictionary`?
It will be very confusing and I actually thought it means directory to start with.

Thanks.

Reply:
No because it's not always a dictionary, it can be `None`, a tuple or a dictionary, so `d` is as good as anything.

Reply:
I think a better naming would be easier to understand and search later as well, but this is just nitpik from me.